dataprep-tools/filecoin/boost_create_deals.py

```diff
@@ -408,50 +463,54 @@ def create_deals(metadata_obj):
 
     return 0
 
-
-# Code below should be agnostic to storage backend
-if __name__ == "__main__":
+def main():
+    """Main function to execute the script."""
     if len(sys.argv) < 2:
-        raise ValueError(
-            "Not enough arguments. usage: ", sys.argv[0], " <epoch> [<deal_type>]"
-        )
-
-    logging.info(
-        "boost create deals version %s " "(boost version: %s).", VERSION, BOOST_VERSION
-    )
+        print("Not enough arguments. usage: ", sys.argv[0], " <epoch> [<deal_type>]", file=sys.stderr)
+        sys.exit(1)
 
     epoch = sys.argv[1]
+    deal_type = sys.argv[2] if len(sys.argv) == 3 else ""
+    logging.info(f"boost create deals version {VERSION} (boost version: {deal_config.boost_version}).")
```

Comment:
```
Traceback (most recent call last):
  File "./boost_create_deals.py", line 516, in <module>
    main()
  File "./boost_create_deals.py", line 474, in main
    logging.info(f"boost create deals version {VERSION} (boost version: {deal_config.boost_version}).")
UnboundLocalError: local variable 'deal_config' referenced before assignment
```


deal_config only initialized in the try block below

Reply:
fixed it.